table/block_based/block_based_table_reader.cc

```diff
@@ -2959,10 +2978,10 @@ bool BlockBasedTableIterator<TBlockIter, TValue>::InitDataBlock() {
         /*get_context=*/nullptr, &lookup_context_, s, prefetch_buffer_.get(),
         /*for_compaction=*/lookup_context_.caller ==
             TableReaderCaller::kCompaction);
-    if (!block_iter_.Valid()) {
+    block_iter_points_to_real_block_ = true;
```

Comment:
why reset `block_iter_points_to_real_block_`?

Reply:
we invalidate the block iter when corruption detected. If not set `block_iter_points_to_real_block_ = true`, upper place will check the status of index iter instead of block iter.